src/app/components/ReduxForm/index.js

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect } from 'react';
+import { reduxForm, getFormValues } from 'redux-form';
+import i18 from 'i18next';
+import { UTButton } from '@widergy/energy-ui';
+import { useHistory } from 'react-router';
+import { connect } from 'react-redux';
+import { bool, func, string, shape } from 'prop-types';
+
+import SurveyActions from 'redux/survey/actions';
+import InputForm from 'app/components/InputForm';
+
+import { inputsArray } from './constants';
+import styles from './styles.module.scss';
+
+const ReduxForm = ({
+  handleSubmit,
+  dispatch,
+  change,
+  savedUserName,
+  currentValues,
+  submitting,
+  invalid,
+  pristine,
+  className
+}) => {
+  const history = useHistory();
+
+  const handleCancel = values => {
+    dispatch(SurveyActions.saveUsername({ name: values?.name, lastName: values?.lastName }));
+    history.goBack();
+  };
+
+  useEffect(() => {
+    if (savedUserName?.name) change('name', savedUserName?.name);
+    if (savedUserName?.lastName) change('lastName', savedUserName?.lastName);
+  }, [JSON.stringify(savedUserName)]);
+
+  return (
+    <form className={className} onSubmit={handleSubmit}>
+      {inputsArray.map(input => (
+        <div className={styles.row}>
+          <InputForm
```

Comment:
Can't you use `Field` directly here? why you need `InputForm`?
```suggestion
          <Field
```